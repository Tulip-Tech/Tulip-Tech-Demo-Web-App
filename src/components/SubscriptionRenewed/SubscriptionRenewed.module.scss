@use 'src/styles/variables';
@use 'src/styles/theme';

.title {
  margin-bottom: 16px;
  font-weight: var(--body-font-weight-bold);
  font-size: 24px;
}

.paragraph {
  //empty
}
