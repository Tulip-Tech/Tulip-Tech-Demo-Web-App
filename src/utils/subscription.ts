import { formatPrice } from '#src/utils/formatting';
import type { Offer, Order } from '#types/checkout';

export const getOfferPrice = (offer: Offer) => formatPrice(offer.customerPriceInclTax, offer.customerCurrency, offer.customerCountry);

export const isSVODOffer = (offer: Offer | Order) => offer.offerId?.[0] === 'S';
