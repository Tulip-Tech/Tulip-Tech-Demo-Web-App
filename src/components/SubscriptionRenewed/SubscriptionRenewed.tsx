import React from 'react';
import { useTranslation } from 'react-i18next';

import styles from './SubscriptionRenewed.module.scss';

import Button from '#components/Button/Button';
import { formatDate, formatPrice } from '#src/utils/formatting';
import type { Subscription } from '#types/subscription';
import type { Customer } from '#types/account';

type Props = {
  customer: Customer;
  subscription: Subscription;
  onClose: () => void;
};

const SubscriptionRenewed: React.FC<Props> = ({ onClose, customer, subscription }: Props) => {
  const { t } = useTranslation('account');
  const date = formatDate(subscription.expiresAt);
  const price = formatPrice(subscription.nextPaymentPrice, subscription.nextPaymentCurrency, customer.country);

  return (
    <div>
      <h2 className={styles.title}>{t('subscription_renewed.title')}</h2>
      <p className={styles.paragraph}>{t('subscription_renewed.message', { date, price })}</p>
      <Button label={t('subscription_renewed.back_to_profile')} onClick={onClose} fullWidth />
    </div>
  );
};

export default SubscriptionRenewed;
