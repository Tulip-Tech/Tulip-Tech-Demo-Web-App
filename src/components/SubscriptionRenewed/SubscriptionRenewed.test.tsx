import React from 'react';
import { render } from '@testing-library/react';

import SubscriptionRenewed from './SubscriptionRenewed';

import customer from '#test/fixtures/customer.json';
import subscription from '#test/fixtures/subscription.json';
import type { Subscription } from '#types/subscription';
import type { Customer } from '#types/account';

describe('<SubscriptionRenewed>', () => {
  test('renders and matches snapshot', () => {
    const { container } = render(<SubscriptionRenewed customer={customer as Customer} subscription={subscription as Subscription} onClose={vi.fn()} />);

    expect(container).toMatchSnapshot();
  });
});
