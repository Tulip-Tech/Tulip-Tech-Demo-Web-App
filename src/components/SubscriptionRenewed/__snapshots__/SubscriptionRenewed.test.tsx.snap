// Vitest Snapshot v1

exports[`<SubscriptionRenewed> > renders and matches snapshot 1`] = `
<div>
  <div>
    <h2
      class="_title_d9af9c"
    >
      subscription_renewed.title
    </h2>
    <p>
      subscription_renewed.message
    </p>
    <button
      class="_button_f8f296 _default_f8f296 _outlined_f8f296 _fullWidth_f8f296"
    >
      <span>
        subscription_renewed.back_to_profile
      </span>
    </button>
  </div>
</div>
`;
